x-pack/plugins/security_solution/server/endpoint/routes/resolver/tree/queries/stats.ts

```diff
@@ -121,26 +186,99 @@ export class StatsQuery {
    * @param client used to make requests to Elasticsearch
    * @param nodes an array of unique IDs representing nodes in a resolver graph
    */
-  async search(client: IScopedClusterClient, nodes: NodeID[]): Promise<Record<string, EventStats>> {
+  async search(
+    client: IScopedClusterClient,
+    nodes: NodeID[],
+    alertsClient: AlertsClient | undefined,
+    includeHits: boolean
+  ): Promise<{ eventStats?: Record<string, EventStats>; alertIds?: string[] }> {
     if (nodes.length <= 0) {
       return {};
     }
 
     const esClient = this.isInternalRequest ? client.asInternalUser : client.asCurrentUser;
+    const alertIndex =
+      typeof this.indexPatterns === 'string' ? this.indexPatterns : this.indexPatterns.join(',');
```

Comment:
Somewhat unrelated, but I feel we should have a function for `getArrayAsString` or something that does this check since we do it all over 😂